File: tests/fixtures/python/basic.py
"""
Python 测试用例: 基础类和函数
"""


class User:
    """用户类"""

    def __init__(self, name: str, email: str):
        self.name = name
        self.email = email

    def validate_email(self) -> bool:
        return "@" in self.email

    def to_dict(self) -> dict:
        return {"name": self.name, "email": self.email}


class Session:
    """会话类"""

    @classmethod
    def create(cls, user: User) -> "Session":
        return cls()

    @staticmethod
    def generate_token() -> str:
        return "token"

    def validate(self, token: str) -> bool:
        return len(token) > 0


def login(username: str, password: str) -> Session:
    """登录函数"""
    return Session()


def logout(session: Session) -> None:
    """登出函数"""
    pass


async def fetch_user(user_id: int) -> User:
    """异步函数"""
    return User("test", "test@example.com")


DEFAULT_TIMEOUT = 30
MAX_RETRIES = 3

File: tests/fixtures/objc/basic.m
// ============================================================
// Objective-C Test: Basic Symbols
// ============================================================

#import <Foundation/Foundation.h>

// ------------------------------------------------------------
// Protocol
// ------------------------------------------------------------
@protocol Identifiable <NSObject>

@required
- (NSString *)identifier;

@optional
- (void)identify;

@end

@protocol Cacheable

- (void)cache;
- (void)invalidate;

@end

// ------------------------------------------------------------
// Class Interface
// ------------------------------------------------------------
@interface User : NSObject <Identifiable>

@property (nonatomic, strong) NSString *userId;
@property (nonatomic, strong) NSString *name;
@property (nonatomic, strong) NSString *email;

- (instancetype)initWithName:(NSString *)name email:(NSString *)email;
- (BOOL)validate;
+ (instancetype)userWithName:(NSString *)name email:(NSString *)email;

@end

// ------------------------------------------------------------
// Class Implementation
// ------------------------------------------------------------
@implementation User

- (instancetype)initWithName:(NSString *)name email:(NSString *)email {
    self = [super init];
    if (self) {
        _name = name;
        _email = email;
        _userId = [[NSUUID UUID] UUIDString];
    }
    return self;
}

+ (instancetype)userWithName:(NSString *)name email:(NSString *)email {
    return [[self alloc] initWithName:name email:email];
}

- (BOOL)validate {
    return self.name.length > 0 && [self.email containsString:@"@"];
}

- (NSString *)identifier {
    return self.userId;
}

@end

// ------------------------------------------------------------
// Category
// ------------------------------------------------------------
@interface User (Display)

- (NSString *)displayName;

@end

@implementation User (Display)

- (NSString *)displayName {
    return [NSString stringWithFormat:@"%@ <%@>", self.name, self.email];
}

@end

// ------------------------------------------------------------
// Another Class
// ------------------------------------------------------------
@interface Address : NSObject

@property (nonatomic, strong) NSString *street;
@property (nonatomic, strong) NSString *city;
@property (nonatomic, strong) NSString *country;

@end

@implementation Address
@end

// ------------------------------------------------------------
// C Functions (also valid in ObjC)
// ------------------------------------------------------------
void greetUser(User *user) {
    NSLog(@"Hello, %@!", user.name);
}

BOOL validateEmail(NSString *email) {
    return [email containsString:@"@"];
}
